karpenter/datadog_checks/karpenter/config_models/validators.py

```diff
@@ -0,0 +1,40 @@
+# (C) Datadog, Inc. 2023-present
+# All rights reserved
+# Licensed under a 3-clause BSD style license (see LICENSE)
+from urllib.parse import urlparse
+
+
+def initialize_instance(values, **kwargs):
+    if 'openmetrics_endpoint' in values:
+        validate_url(
+            values['openmetrics_endpoint'],
+            required_path='/metrics',
+            example='http://localhost:8000/metrics',
+            config='openmetrics_endpoint',
+        )
+
+    return values
+
+
+def validate_url(url, required_path=None, example=None, config=None):
+    # Validates the endpoint to ensure that the components are presents.
+    # For OpenMetrics: Scheme, netloc and /metrics path
+    # For API: Scheme, netloc and no path
```

Comment:
Seems like this line is a copy/paste from weaviate 🙂

Reply:
Yes. I copied it from there because I thought it would be helpful